native/cocos/ui/edit-box/EditBox-ios.mm

```diff
@@ -363,76 +267,331 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    InputBoxPair* ret;
+    CGRect viewRect = UIApplication.sharedApplication.delegate.window.rootViewController.view.frame;
+    
+    //TODO: object for key with real hash value
+    NSString* inputType = getTextInputType(showInfo);
+    
+    if ((ret = [textInputDictionnary objectForKey:inputType])) {
+        [[ret inputOnView] setFrame:CGRectMake(showInfo->x,
+                                 viewRect.size.height - showInfo->y - showInfo->height,
+                                 showInfo->width,
+                                 showInfo->height)];
+        CGRect safeArea = getSafeAreaRect();
+        [[[ret inputOnView] inputAccessoryView] setFrame:CGRectMake(0,
+                                                                           0,
+                                                                    safeArea.size.width,
+                                                                    getTextInputHeight(g_isMultiline) + ITEM_MARGIN_HEIGHT)];
+        [self setInputWidthOf:[[ret inputOnView] inputAccessoryView] ];
+    } else {
+        ret = [[InputBoxPair alloc] init];
+        [ret setInputOnView:[[UITextField alloc]
+                initWithFrame:CGRectMake(showInfo->x,
+                                         viewRect.size.height - showInfo->y - showInfo->height,
+                                         showInfo->width,
+                                         showInfo->height)]];
+        [textInputDictionnary setValue:ret forKey:inputType];
+        [self addInputAccessoryViewForTextField:ret with:showInfo];
+    }
+    ((UITextField*)[ret inputOnToolbar]).text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
+    ((UITextField*)[ret inputOnView]).text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
+    setTextFieldReturnType((UITextField*)[ret inputOnToolbar], showInfo->confirmType);
+    setTextFieldKeyboardType((UITextField*)[ret inputOnToolbar], showInfo->inputType);
+    return ret;
 }
-@end
 
+//TODO: show inputbox with width modified.
+- (void) show: (const cc::EditBox::ShowInfo*)showInfo {
+    g_maxLength = showInfo->maxLength;
+    g_isMultiline = showInfo->isMultiline;
+    
+    if (g_isMultiline) {
+        curView = [self createTextView:showInfo];
+    } else {
+        curView = [self createTextField:showInfo];
+    }
+    [[curView inputOnView] setHidden:INPUTBOX_HIDDEN];
+    UIView *view = UIApplication.sharedApplication.delegate.window.rootViewController.view;
+    
+    [view addSubview:[curView inputOnView]];
+    [[curView inputOnView] becomeFirstResponder];
+    [[curView inputOnToolbar] becomeFirstResponder];
+}
+// Change the focus point to the TextField or TextView on the toolbar.
+- (void) hide {
+    [[curView inputOnView] becomeFirstResponder];
+    [[curView inputOnView] removeFromSuperview];
+    [[curView inputOnToolbar] resignFirstResponder];
+    [[curView inputOnView] resignFirstResponder];
```

Comment:
Should release curView here?

Reply:
No! It's week ref here, use it but not release it.